### DISCUSSION TITLE
Authentication Question

### DISCUSSION THREAD

**Discussion by cwmn** [homebridge-simplisafe3/homebridge-simplisafe3#275] [Q&A] [answered]
I followed the instructions for copying the new authentication url, but it's not clear to me where to paste this in the config file? I realize this is probably a stupid question but....

**Comment by cwmn** [comment]
Ok, my mistake! The initial issue is that I was to get the authentication from my laptop, not the desktop with Homebridge installed, sorry. But even when attempting to authenticate on my HB cpu I'm having issues. Note my HB server is an older computer run-in Mac OS 10.11. Many browsers complain that they can't get a secure connection with Simplisafe, if either use the terminal or bypass the security warnings I get to the enter username/password then upon entering the security code that is texted me clicking OK does absolutely nothing! Any ideas of how to make this work short of installing on a newer computer?

**Comment by cwmn** [comment]
Thanks for the note, with Brave and Firefox I get all the way to the box asking me for the text authentication, which I receive but after entering the code the Continue button doesn’t do a thing, just sits there. Any other suggestions?

Thanks

On Jan 2, 2022, at 4:54 PM, Michael Shamoon ***@***.***> wrote:


It doesn’t matter what machine you use for the process, you just need to be able to access the homebridge server via browser or terminal. You could copy the link from one machine, open it up on another and then paste the generated URL back to the HB machine. Only thing you can’t use is mobile. Please see the notes in the readme, in some browsers the URL is shown in the error console rather than redirecting you to it.

authentication is no longer saved in the config, you need to let the Homebridge UI or the terminal command generate an auth file.

—
Reply to this email directly, view it on GitHub <https://github.com/homebridge-simplisafe3/homebridge-simplisafe3/discussions/275#discussioncomment-1897430>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AD3N5Z37FH5N75YTFDRBOELUUDJRJANCNFSM5LD6BQCQ>.
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by shamoon** [reply]
As I mentioned and is in the readme, please check the browser error console, FF for example wont redirect you, it will just output the URL you need in the console.

**Comment by shamoon** [comment, marked as answer]
It doesn’t matter what machine you use for the process, you just need to be able to access the homebridge server via browser or terminal. You could copy the link from one machine, open it up on another and then paste the generated URL back to the HB machine. Only thing you can’t use is mobile. Please see the notes in the readme, in some browsers the URL is shown in the error console rather than redirecting you to it.

authentication is no longer saved in the config, you need to let the Homebridge UI or the terminal command generate an auth file.

**Comment by cwmn** [comment]
Sorry to be a pain, but I find the readme and find it just a bit lacking, but your messages are helping, thanks. Possible light bulb moment, when I put in the two factor authentication and it goes nowhere I see this in the console Failed to launch 'com.simplisafe.mobile://auth.simplisafe.com/ios/com.simplisafe.mobile/callback?code=TgS1jf-Psp3zK' because the scheme does not have a registered handler. Is the highlighted text what I’m looking for? Also can I paste that into the Redirected URL on any machine or must I be connected to the HB server?

Thanks,
Chuck

On Jan 2, 2022, at 5:40 PM, Michael Shamoon ***@***.***> wrote:


It doesn’t matter what machine you use for the process, you just need to be able to access the homebridge server via browser or terminal. You could copy the link from one machine, open it up on another and then paste the generated URL back to the HB machine. Only thing you can’t use is mobile. Please see the notes in the readme, in some browsers the URL is shown in the error console rather than redirecting you to it.

authentication is no longer saved in the config, you need to let the Homebridge UI or the terminal command generate an auth file.

—
Reply to this email directly, view it on GitHub <https://github.com/homebridge-simplisafe3/homebridge-simplisafe3/discussions/275#discussioncomment-1897533>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AD3N5Z5MJFX55YHRYIBCOPDUUDO6JANCNFSM5LD6BQCQ>.
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by shamoon** [reply]
Its not the most obvious process but I thought it was covered in the readme. From the readme:
> launch the authentication process and you will have to **copy and paste the final URL (begins with com.SimpliSafe.mobile://) back into the plugin settings**.

And:
> Note that in some browsers (e.g. Chrome) the browser will not redirect you and will show an error in the Console (e.g. View > Developer Tools > Javascript Console) and you will have to copy and paste the URL from the error message.

So yes, copy and paste that _com.SimpliSafe..._ URL back into the settings.

**Comment by cwmn** [comment]
It appears to be working! FYI, I probably will have everyone slapping their head when they see the last post with the url or my token, I did modify the URL before sending, so no worries. Anyway thank for the help, sorry to seem so dense.

Chuck

On Jan 2, 2022, at 5:40 PM, Michael Shamoon ***@***.***> wrote:


It doesn’t matter what machine you use for the process, you just need to be able to access the homebridge server via browser or terminal. You could copy the link from one machine, open it up on another and then paste the generated URL back to the HB machine. Only thing you can’t use is mobile. Please see the notes in the readme, in some browsers the URL is shown in the error console rather than redirecting you to it.

authentication is no longer saved in the config, you need to let the Homebridge UI or the terminal command generate an auth file.

—
Reply to this email directly, view it on GitHub <https://github.com/homebridge-simplisafe3/homebridge-simplisafe3/discussions/275#discussioncomment-1897533>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AD3N5Z5MJFX55YHRYIBCOPDUUDO6JANCNFSM5LD6BQCQ>.
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by shamoon** [reply]
Ha, no worries. Because of restrictions based off how SS have implemented authentication this is about as user-friendly as we can make the process, not ideal tho.

**Comment by cwmn** [comment]
I think it’s the case of overthinking or worse yet making assumptions. I assumed that the two factor authentication was needed and that then the url would break. The fact that the Continue button on the verification didn’t seem to do anything threw me for a loop. Thanks for all the assistance.

Chuck


On Jan 2, 2022, at 6:07 PM, Michael Shamoon ***@***.***> wrote:


Its not the most obvious process but I thought it was covered in the readme. From the readme:

launch the authentication process and you will have to copy and paste the final URL (begins with com.SimpliSafe.mobile://) back into the plugin settings.

And:

Note that in some browsers (e.g. Chrome) the browser will not redirect you and will show an error in the Console (e.g. View > Developer Tools > Javascript Console) and you will have to copy and paste the URL from the error message.

So yes, copy and paste that com.SimpliSafe... URL back into the settings.

—
Reply to this email directly, view it on GitHub <https://github.com/homebridge-simplisafe3/homebridge-simplisafe3/discussions/275#discussioncomment-1897581>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AD3N5Z4ZEXGSTZ5HUCYOPM3UUDSFTANCNFSM5LD6BQCQ>.
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by Homekitdude** [comment]
Can anyone help me in doing this via iPad Pro (or MacBook Pro) via HOOBS?

**Reply by shamoon** [reply]
HOOBS is a commercial product not directly affiliated with this plug-in, I believe the developers of HOOBS are responsible for ensuring it works with this so if you’re having a problem you may want to contact them. But in general the instructions are basically the same, I think.

**Comment by Homekitdude** [comment]
Man I just feel really dumb, but the README section says something like “press the button” and it’ll take me to the link to copy but it’s so vague and I’m lost as I’m not a coder or good with any of this stuff but instead very basic.

**Reply by shamoon** [reply]
Attached is a screenshot when using Homebridge which is what I build and test this plugin with, I dont have HOOBS but as I understand it they update their software to work with Homebridge plugins. Again, you may want to reach out to them since you paid for a product that relies on free open source software I should hope they provide support.
<img width="814" alt="Screen Shot 2022-01-03 at 11 45 46 AM" src="https://user-images.githubusercontent.com/4887959/147973488-f95edb63-ac8e-45ac-96fd-3c28577bddad.png">

**Comment by cwmn** [comment]
Attached is a pdf of what I did to get Simplisafe configured.

Regards,
Chuck



On Jan 3, 2022, at 11:58 AM, Homekitdude ***@***.***> wrote:


Man I just feel really dumb, but the README section says something like “press the button” and it’ll take me to the link to copy but it’s so vague and I’m lost as I’m not a coder or good with any of this stuff but instead very basic.

—
Reply to this email directly, view it on GitHub <https://github.com/homebridge-simplisafe3/homebridge-simplisafe3/discussions/275#discussioncomment-1901425>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AD3N5Z6Z4HXHOVXQQQJ2VXDUUHPVBANCNFSM5LD6BQCQ>.
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by Homekitdude** [reply]
I'm sorry, but where is the PDF?

**Comment by Homekitdude** [comment]
All I have the option to do on the desktop HOOBs app is to unlink account.
![Screen Shot 2022-01-03 at 2 37 54 PM](https://user-images.githubusercontent.com/96741140/147977902-7c58772e-7b97-44fa-a0f0-a63b897f0091.png)

**Reply by shamoon** [reply]
@cwmn , Homwkitdude is using HOOBS, not homebride though thanks for the effort.

@Homekitdude, as I said, they are not affiliated with this plugin at all, they just update their software to work with it as needed. So I think you should contact HOOBS. Or just use Homebridge =)

**Comment by cwmn** [comment]
Perhaps attachments get stripped from emails, I’ll try again and add the information below as well. Hopefully the photos won’t get removed as well.






I just went through this yesterday so I will try to help you. Disclaimer: I’m running HomeBridge and not familiar with any differences with HOOBS.

 

I’m on a Mac Book Pro, do not do this with your iPad or phone use a computer.  In the end my issue was created mainly by misinterpreting the ReadMe instructions, I hope this helps to clarify the process.

Log in as admin on your Homebridge site from a computer
Go to the Plugins tab
Click on Homebridge Simplisafe3 Settings
Click on the purple LAUNCH SIMPLISAFE LOGIN
Simplisafe will open your default browser, enter your username and password and click Continue
A dialog box will open asking you to verify your identity. In addition a six digit code will be sent to you cell phone, or the phone which is designated as your SimpliSafe contact number. When you get the text enter the number in the dialog box and click Continue.

This is where I got very confused, because the continue button didn’t seem to do anything. I tried Safari, Firefox, Chrome, Opera and Brave. The secret is you don’t want it to do anything, you are trying to capture the link the Continue button is trying to connect to as it has your personal authentication information that you need for your plugin config.

In Safari the information you need is provided in the browser window like this, the information highlighted between the quotes is the authentication configuration that you need for your plugin. If you have Safari configured to show the complete URL in the address bar you may be able to copy the address there to use.

·      If you are using a different browser you will need to make sure the Developer Tools are enabled in the preferences and after submitting the text string verifying your identity, you will need to go to the Developer Tools and open the Java Console as shown here:

 


 

Once the console is open you should see the url that you need to paste into the plugin configuration, as shown below. Note I didn’t put in an actual code, which is why you see an error message but you get the idea right? 



 

Simply copy the highlighted code as shown below and paste it in the Redirected URL blank as demonstrated in the image below.

 


 

As you can see in the lower right there is an error as my code is totally bogus. But a legitimate code will verify after a moment and you should be good to go.

 

Good luck!

**Reply by Homekitdude** [reply]
Thanks for this but it seems to be different in HOOBs. I actually just unlinked and relinked my account and it resolved my issues. I was worried I'd have to re-add all my sensors to their respective rooms but that wasnt the case.

**Comment by cwmn** [comment]
I knew that but don’t know a thing about HOOBS, thought it was a commercialized HB and that the instructions might help. Bottom line the detailed instructions might help someone else down the line. I wasted a few hours of my and your time yesterday. Thought a bit more detail might be worth it for others.

On Jan 3, 2022, at 3:05 PM, Michael Shamoon ***@***.***> wrote:


@cwmn <https://github.com/cwmn> , Homwkitdude is using HOOBS, not homebride though thanks for the effort.

@Homekitdude <https://github.com/Homekitdude>, as I said, they are not affiliated with this plugin at all, they just update their software to work with it as needed. So I think you should contact HOOBS. Or just use Homebridge =)

—
Reply to this email directly, view it on GitHub <https://github.com/homebridge-simplisafe3/homebridge-simplisafe3/discussions/275#discussioncomment-1902273>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AD3N5Z5XUJX3U2NMIYUXAD3UUIFQZANCNFSM5LD6BQCQ>.
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
